dev/RadioButtons/RadioButtons.cpp

```diff
@@ -547,6 +561,58 @@ void RadioButtons::UpdateSelectedItem()
     }
 }
 
+winrt::ToggleButton RadioButtons::TryGetRadioButtonFromRepeaterElement(const winrt::UIElement& element)
+{
+    if (auto const elementAsGrid = element.try_as<winrt::Grid>())
+    {
+        if (elementAsGrid.Children().Size() == 1)
+        {
+            return elementAsGrid.Children().GetAt(0).try_as<winrt::RadioButton>();
+        }
+    }
+
+    return nullptr;
+}
+
+winrt::ToggleButton RadioButtons::TryGetRadioButtonFromRepeaterElementIndex(int index, const winrt::ItemsRepeater& repeater)
+{
+    if (auto const repeaterElement = repeater.TryGetElement(index))
+    {
+        return TryGetRadioButtonFromRepeaterElement(repeaterElement);
+    }
+
+    return nullptr;
+}
+
+winrt::UIElement RadioButtons::TryGetRepeaterElementFromRadioButton(const winrt::RadioButton& radioButton)
+{
+    return winrt::VisualTreeHelper::GetParent(radioButton).try_as<winrt::Grid>();
```

Comment:
>Grid [](start = 73, length = 4)

Should this be try_as<winrt::UIElement>? It seems like this code is making an assumption about the shape of the Item template, does this work if a custom one is provided?

Reply:
In our WinUI implementation, we know that a RepeaterElement is a Grid (since we enforce that in the RepeaterElementFactory implementation). 

I guess, however, that a developer could supply a custom ItemTemplate to the ItemsRepeater in use and they might not choose a Grid as the parent element. Using UIElement instead seems like the better choice.